### ISSUE TITLE
Installing a specific version of a mod that requires incompatibility confirmation should select that exact version by default

### ISSUE THREAD

**Issue by Naxum** [modrinth/code#2476]
### Please confirm the following.

- [X] I checked the [existing issues](https://github.com/modrinth/code/issues) for duplicate problems
- [X] I have tried resolving the issue using the [support portal](https://support.modrinth.com)
- [X] I have ensured my Modrinth App installation is up to date

### What version of the Modrinth App are you using?

0.8.7

### What operating systems are you seeing the problem on?

Windows

### Describe the bug

I'm using NeoForged on Minecraft 1.20.1 which is an unfortunate transitionary period where many mods added official NeoForged support in 1.21, so their 1.20 versions are listed as "incompatible" despite Forge apps working in NeoForged for my version.

Hitting "update all" solely updates the mods with explicit NeoForged support. So, I have to go through my mods and manually select their most recent version after filtering for Forge AND NeoForged. My main issue is when selecting a specific exact version via its "install" button on its detail page or the quick swap button both open an "incompatibility warning" requiring a selection of a version to continue.

The issue: the selected version is NOT the one you just manually navigated to and selected. Instead you have to browse the menu without filtering ability. 

List of mod versions:
![image](https://github.com/user-attachments/assets/44f83b03-feb8-4c2e-ab7c-04f17f31f6d8)

Trying to install a specific version (24.4.0 for Forge 1.20.1) instead asks if you want to install seemingly a random choice. (24.4.0 for Forge 1.16.x)
![image](https://github.com/user-attachments/assets/6f2b0b85-7fa9-4133-b060-8f8daba09c45)

The menu is painful to browse.
![image](https://github.com/user-attachments/assets/948ccaf1-3226-45b4-9f1a-2a1411c5194e)

Instead, I open this version on the website so I can download the zip manually there and drag it into the folder.

### Steps to reproduce

1. Create a profile for NeoForged 1.20.1
2. Download a mod like Xaero's Minimap that doesn't explicitly support NeoForged for 1.20.1
3. Navigate to that mod's listed versions
4. Note that you may need to choose both "Forge" & "NeoForged" in the filter list
5. Attempt to install another specific version
6. Observe compatibility sheet does not have that version selected

### Expected behavior

I expected the confirmation dialog to have the version I just came from selected so I can simply continue with a single button press.

**Comment by potential1energy** [comment]
Yes this has been very annoying when installing forge mods for neoforge, there's no filtering for the incompatibility select, and it should select the file I tried to download by default, why is this not a feature??

**Comment by X66Herobrine66X** [comment]
+1, would love to see this fixed. I'm honestly not sure if this is technically a bug or just an oversight on the part of the UI devs.

**Comment by T1xx1** [comment]
+1

**Comment by bitgen-was-taken** [comment]
please fix this it is SO ANNOYING
